apps/service_providers/llm_service/runnables.py

```diff
@@ -248,12 +250,8 @@ def _build_chain(self) -> Runnable[dict[str, Any], Any]:
 
 class AgentLLMChat(LLMChat):
     def _parse_output(self, output):
-        output = output.get("output", "")
-        if isinstance(output, list):
-            # Responses API responses are lists
-            return "\n".join([o["text"] for o in output])
-
-        return output
+        output_parser = self.adapter.get_llm_service().get_output_parser()
+        return output_parser(output)
```

Comment:
I think `_get_cited_files` should also be provider specific like `get_output_parser` is